povm_toolbox/quantum_info/multi_qubit_frame.py

```diff
@@ -179,22 +237,25 @@ def __array__(self) -> np.ndarray:
     def analysis(
         self,
         hermitian_op: SparsePauliOp | Operator,
-        frame_op_idx: int | set[int] | None = None,
-    ) -> float | dict[int, float] | np.ndarray:
+        frame_op_idx: LabelMultiQubitT | set[LabelMultiQubitT] | None = None,
+    ) -> float | dict[LabelMultiQubitT, float] | np.ndarray:
         if isinstance(hermitian_op, SparsePauliOp):
             hermitian_op = hermitian_op.to_operator()
         op_vectorized = np.conj(matrix_to_double_ket(hermitian_op.data))
 
-        if isinstance(frame_op_idx, int):
-            return float(np.dot(op_vectorized, self._array[:, frame_op_idx]).real)
+        if isinstance(frame_op_idx, (int, tuple)):
+            return float(
+                np.dot(op_vectorized, self._array[:, self._ravel_index(frame_op_idx)]).real
+            )
         if isinstance(frame_op_idx, set):
             return {
-                idx: float(np.dot(op_vectorized, self._array[:, idx]).real) for idx in frame_op_idx
+                idx: float(np.dot(op_vectorized, self._array[:, self._ravel_index(idx)]).real)
+                for idx in frame_op_idx
             }
         if frame_op_idx is None:
             return np.array(np.dot(op_vectorized, self._array).real)
         raise TypeError(
-            "The optional `frame_op_idx` can either be a single or set of integers, not a "
+            "The optional ``frame_op_idx`` can either be a single or set of integers, not a "
```

Comment:
In error messages we do not need to use RST syntax and I prefer to stick to markdown as this is known by more users.
```suggestion
            "The optional `frame_op_idx` can either be a single or set of integers, not a "
```